src/Employer/Employer.Web/Controllers/EditVacancyDatesController.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.AspNetCore.Mvc;
+using Esfa.Recruit.Employer.Web.Configuration;
+using Microsoft.AspNetCore.Hosting;
+using System;
+
+namespace Esfa.Recruit.Employer.Web.Controllers
+{
+    [Route(RoutePrefixPaths.AccountVacancyRoutePath)]
+    public class EditVacancyDatesController : Controller
+    {
+        private readonly EditVacancyDatesOrchestrator _orchestrator;
+        private readonly IHostingEnvironment _hostingEnvironment;
+
+        public EditVacancyDatesController(EditVacancyDatesOrchestrator orchestrator, IHostingEnvironment hostingEnvironment)
+        {
+            _orchestrator = orchestrator;
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        [HttpGet("edit-dates", Name = RouteNames.VacancyEditDates_Get)]
+        public async Task<IActionResult> EditVacancyDates(VacancyRouteModel vrm)
+        {
+            string proposedClosingDate = Request.Cookies[string.Format(CookieNames.VacancyProposedClosingDate, vrm.VacancyId)]?.Trim();
+            string proposedStartDate = Request.Cookies[string.Format(CookieNames.VacancyProposedStartDate, vrm.VacancyId)]?.Trim();
+
+            var vacancy = await _orchestrator.GetVacancyAsync(vrm);
+
+            if (vacancy.CanEdit)
+            {
+                return HandleRedirectOfDraftVacancy(vacancy);
+            }
+
+            if (vacancy.Status != VacancyStatus.Live)
+            {
+                return RedirectToRoute(RouteNames.DisplayVacancy_Get);
+            }
+
+            var response = await _orchestrator.GetEditVacancyDatesViewModel(vrm, proposedClosingDate, proposedStartDate);
+
+            if (!response.Success)
+            {
+                response.AddErrorsToModelState(ModelState);
+            }
+
+            return View(response.Data);
+        }
+
+        [HttpPost("edit-dates", Name = RouteNames.VacancyEditDates_Post)]
+        public async Task<IActionResult> EditVacancyDates(EditVacancyDatesEditModel m)
+        {
+            var vacancy = await _orchestrator.GetVacancyAsync(m);
+
+            if (vacancy.CanEdit)
+            {
+                return HandleRedirectOfDraftVacancy(vacancy);
+            }
+
+            if (vacancy.Status != VacancyStatus.Live)
+            {
+                return RedirectToRoute(RouteNames.DisplayVacancy_Get);
+            }
+
+            var response = await _orchestrator.ValidateEditVacancyDatesViewModel(m);
+
+            if (!response.Success)
+            {
+                response.AddErrorsToModelState(ModelState);
+            }
+
+            if (!ModelState.IsValid)
+            {
+                var vm = await _orchestrator.GetEditVacancyDatesViewModel(m);
+                return View(vm);
+            }
+
+            Response.Cookies.Append(string.Format(CookieNames.VacancyProposedClosingDate, m.VacancyId), DateTime.Parse(m.ClosingDate).ToShortDateString(), EsfaCookieOptions.GetShortLifeHttpCookieOption(_hostingEnvironment));
```

Comment:
Is it possible to have just one cookie and serialize a DTO that contains the dates as DateTime properties? Maybe even store the vacancyId in the DTO so we can be sure the value pertain to the right vacancy. In future we could add further proposed changes to this cookie.

Reply:
Would prefer to avoid serializing and deserailizing json for the moment.